### PR TITLE
[LM-9624] LMMedia

### PR DESCRIPTION
**Pull request by prabha-likeminds**
## Description

**Features**
- Added LMImage, LMVideo, LMCarousel, LMDocument, LMLinkPreview components

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: src/base/LMLoader/types.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface LoaderProps {
```
Comment (ishaan1607) [inline src/base/LMLoader/types.ts]: rename to `LMLoaderProps`

File: src/base/LMLoader/index.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import {ActivityIndicator, View} from 'react-native';
+import React from 'react';
+import { LoaderProps } from './types';
+
+const LMLoader = ({color, size}: LoaderProps) => {
+  return <ActivityIndicator size={size ? size : 'large'} color={color? color : '#5046E5'} />;
```
Comment (ishaan1607) [inline src/base/LMLoader/index.tsx]: we should `#5046E5` to some variable and then use it.

File: src/components/LMMedia/LMCarousel/index.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      style={StyleSheet.flatten([styles.swiperView, carouselStyle])}
+      // handling custom style of pagination container
+      paginationStyle={StyleSheet.flatten([
+        styles.paginationView,
+        paginationBoxStyle,
+      ])}
+      // handling custom style of active pagination item
+      paginationStyleItemActive={StyleSheet.flatten([
+        styles.paginationItemStyle,
+        {backgroundColor: '#5046E5'},
```
Comment (ishaan1607) [inline src/components/LMMedia/LMCarousel/index.tsx]: we should `#5046E5` to some variable and then use it.

File: src/components/LMMedia/LMCarousel/index.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        activeIndicatorStyle,
+      ])}
+      // handling custom style of inactive pagination item
+      paginationStyleItemInactive={StyleSheet.flatten([
+        styles.paginationItemStyle,
+        {backgroundColor: '#e0e0e0'},
```
Comment (ishaan1607) [inline src/components/LMMedia/LMCarousel/index.tsx]: we should `#e0e0e0` to some variable and then use it.

File: src/components/LMMedia/LMCarousel/types.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import {ViewStyle} from 'react-native';
+import {LMImageProps} from '../LMImage/types';
+import {LMVideoProps} from '../LMVideo/types';
+
+export interface LMCarouselProps {
+  attachments: Array<object>; // this represents the array of image & video attachments to be displayed
```
Comment (ishaan1607) [inline src/components/LMMedia/LMCarousel/types.ts]: this should be `Array<LMAttachmentUI>` and not `object`, these are UI models we are talking about.

File: src/components/LMMedia/LMDocument/index.tsx [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+                    {
+                      display:
+                        showDocumentSize != undefined
+                          ? showDocumentSize
+                            ? 'flex'
+                            : 'none'
+                          : 'flex',
+                    },
+                  ])}>
+                  278 Kb
```
Comment (ishaan1607) [inline src/components/LMMedia/LMDocument/index.tsx:124]: Why we have added this `278 kb`, if it is a dummy value, please add a todo

File: src/components/LMMedia/LMDocument/index.tsx [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+                    {
+                      display:
+                        showPageCount != undefined
+                          ? showPageCount
+                            ? 'flex'
+                            : 'none'
+                          : 'flex',
+                    },
+                  ])}>
+                  2 Pages
```
Comment (ishaan1607) [inline src/components/LMMedia/LMDocument/index.tsx:91]: Why we have added this `278 kb`, if it is a dummy value, please add a todo

File: src/components/LMMedia/LMDocument/index.tsx [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+  },
+  alignRow: {
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  docTitle: {
+    color: '#00000099',
+    fontSize: 18,
+    fontWeight: '500',
+    // fontFamily: STYLES.$FONT_FAMILY.REGULAR
```
Comment (ishaan1607) [inline src/components/LMMedia/LMDocument/index.tsx]: remove commented code?

File: src/components/LMMedia/LMDocument/index.tsx [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+  },
+  docDetail: {
+    color: '#00000077',
+    fontSize: 14,
+    // fontFamily: STYLES.$FONT_FAMILY.REGULAR
```
Comment (ishaan1607) [inline src/components/LMMedia/LMDocument/index.tsx]: remove commented code

File: src/components/LMMedia/LMDocument/types.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+  ogTags: Object;
+  size: number;
+  url: string;
+}
+
+// data model for array of attachments
+interface DocumentItem {
+  attachmentMeta: AttachmentMeta;
+  attachmentType: number;
+}
```
Comment (ishaan1607) [inline src/components/LMMedia/LMDocument/types.ts]: move this interface to separate folder and use builder pattern to make these models.

File: src/components/LMMedia/LMDocument/types.ts [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+  entityId: string;
+  format: string;
+  name: string;
+  ogTags: Object;
+  size: number;
+  url: string;
+}
+
+// data model for array of attachments
+interface DocumentItem {
```
Comment (ishaan1607) [inline src/components/LMMedia/LMDocument/types.ts]: this will LMAttachmentUI

File: src/components/LMMedia/LMDocument/types.ts [hunk 12]
```diff
@@ -0,0 +1,31 @@
+import {TextStyle, ViewStyle} from 'react-native';
+import {LMIconProps} from '../../../base/LMIcon/types';
+
+// data model for attachmentMeta object inside attachments
+interface AttachmentMeta {
```
Comment (ishaan1607) [inline src/components/LMMedia/LMDocument/types.ts]: this will be LMAttachmentMetaUI

File: src/components/LMMedia/LMImage/index.tsx [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+        ])}
+      />
+      {/* this renders the error whenever the media is not fetched */}
+      {error ? (
+        <View style={[defaultStyles.imageStyle, defaultStyles.errorView]}>
+          {errorWidget ? (
+            errorWidget
+          ) : (
+            <Text style={defaultStyles.errorText}>
+              An error occured while fetching the media
```
Comment (ishaan1607) [inline src/components/LMMedia/LMImage/index.tsx]: move this string to constant file.

File: src/components/LMMedia/LMLinkPreview/index.tsx [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+  linkImageStyle,
+  linkTitleStyle,
+  linkDescriptionStyle,
+  linkUrlStyle,
+}: LMLinkPreviewProps) => {
+  return (
+    <TouchableOpacity
+      style={styles.postMedia}
+      onPress={() => {
+        Linking.openURL(attachments?.attachmentMeta?.ogTags?.url), onTap;
```
Comment (ishaan1607) [inline src/components/LMMedia/LMLinkPreview/index.tsx]: assign `attachments?.attachmentMeta?.ogTags` to a variable and the use this.

File: src/components/LMMedia/LMLinkPreview/types.ts [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+  title: string;
+  url: string;
+  image?: string;
+}
+
+// data model for attachmentMeta
+interface AttachmentMeta {
+  entityId: string;
+  ogTags: OgTagsProps;
+}
```
Comment (ishaan1607) [inline src/components/LMMedia/LMLinkPreview/types.ts]: move this interface to different folder, and then use it here and use builder pattern for creating this class.

File: src/components/LMMedia/LMLinkPreview/types.ts [hunk 16]
```diff
@@ -0,0 +1,29 @@
+import {ImageStyle, TextStyle, ViewStyle} from 'react-native';
+
+// data model for ogTags object in attachmentMeta
+interface OgTagsProps {
```
Comment (ishaan1607) [inline src/components/LMMedia/LMLinkPreview/types.ts]: this will be LMOGTagsUI

File: src/components/LMMedia/LMLinkPreview/types.ts [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+  description: string;
+  title: string;
+  url: string;
+  image?: string;
+}
+
+// data model for attachmentMeta
+interface AttachmentMeta {
```
Comment (ishaan1607) [inline src/components/LMMedia/LMLinkPreview/types.ts]: repeated class

File: src/components/LMMedia/LMVideo/index.tsx [hunk 18]
```diff
@@ -0,0 +1,185 @@
...
+      ) : null}
+
+      {/* this shows a error if the media is not fetched */}
+      {error ? (
+        <View style={[defaultStyles.videoStyle, defaultStyles.errorView]}>
+          {errorWidget ? (
+            errorWidget
+          ) : (
+            <Text style={defaultStyles.errorText}>
+              An error occured while fetching the media
```
Comment (ishaan1607) [inline src/components/LMMedia/LMVideo/index.tsx]: move this string to a constant and then use it.

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

**Review by ishaan1607 — APPROVED** [review]

**Review by divyanshgandhilm — APPROVED** [review]

**Pull request merged**